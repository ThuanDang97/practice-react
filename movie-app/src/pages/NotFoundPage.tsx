import { Link } from 'react-router-dom';
import '../styles/error404.css';

const NotFoundPage = (): JSX.Element => {
  return (
    <>
      <h1>Page Not Found</h1>
      <p className="zoom-area">
        <b>Oops!</b> Something is wrong.{' '}
      </p>
      <section className="error-container">
        <span className="four">
          <span className="screen-reader-text">4</span>
        </span>
        <span className="zero">
          <span className="screen-reader-text">0</span>
        </span>
        <span className="four">
          <span className="screen-reader-text">4</span>
        </span>
      </section>
      <div className="link-container">
        <Link to="/" className="more-link">
          Go HomePage
        </Link>
      </div>
    </>
  );
};

export { NotFoundPage };
