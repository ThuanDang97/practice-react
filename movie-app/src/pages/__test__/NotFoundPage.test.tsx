import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { NotFoundPage } from '../NotFoundPage';

describe('Test Not Found Page', () => {
  test('Not Found Page matches DOM Snapshot', () => {
    const errorPage = (
      <Router>
        <NotFoundPage />;
      </Router>
    );
    render(errorPage);

    expect(errorPage).toMatchSnapshot();
  });
});
