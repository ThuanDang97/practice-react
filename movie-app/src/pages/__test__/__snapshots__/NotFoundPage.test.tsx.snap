// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Not Found Page Not Found Page matches DOM Snapshot 1`] = `
<BrowserRouter>
  <NotFoundPage />
  ;
</BrowserRouter>
`;
